File: app/src/main/java/threadpoolexec/com/threadpoolexecutor/services/LocalBinder.java
package threadpoolexec.com.threadpoolexecutor.services;

/**
 * Custom Binder class
 */

import android.os.Binder;

import java.lang.ref.WeakReference;

public class LocalBinder<S> extends Binder {
    private final WeakReference<S> mService;

    public LocalBinder(final S service) {
        mService = new WeakReference<S>(service);
    }

    public S getService() {
        return mService.get();
    }
}

File: app/src/main/java/threadpoolexec/com/threadpoolexecutor/services/LocalBinder.kt
package threadpoolexec.com.threadpoolexecutor.services

/**
 * Custom Binder class
 */

import android.os.Binder

import java.lang.ref.WeakReference

class LocalBinder<S>(service: S) : Binder() {
    private val mService: WeakReference<S> = WeakReference(service)

    val service: S?
        get() = mService.get()

}